app/controllers/application_controller.rb

```diff
@@ -11,4 +11,39 @@ def configure_permitted_parameters
     # For additional in app/views/devise/registrations/edit.html.erb
     devise_parameter_sanitizer.permit(:account_update, keys: [:username, :nickname, :email])
   end
+
+  def current_or_guest_user
+    if current_user
+      if session[:guest_user_id] && session[:guest_user_id] != current_user.id
+        # logging_in no need for now
+        # reload guest_user to prevent caching problems before destruction
+        # guest_user(with_retry = false).try(:reload).try(:destroy)
```

Comment:
Commented this line out, because it tries to delete the guest user and activerecord doesn't like that as instance should always have a user registered.